.travis.yml

```diff
@@ -28,7 +28,9 @@ install:
   - if [[ $setup = 'stable' ]]; then travis_retry composer update --prefer-dist --no-interaction --no-suggest --prefer-stable; fi
   - if [[ $setup = 'lowest' ]]; then travis_retry composer update --prefer-dist --no-interaction --no-suggest --prefer-stable --prefer-lowest; fi
 
-script: vendor/bin/phpunit --verbose --coverage-text --coverage-clover=coverage.xml
+script:
+  - vendor/bin/phpunit --verbose --coverage-text --coverage-clover=coverage.xml
+  - if [[ $PHPSTAN = 1 ]]; then composer require --dev phpstan/phpstan:^0.5.2 && vendor/bin/phpstan analyse -c phpstan.neon -l 3 src tests; fi
```

Comment:
I will use the long arg notation.

Level 4 is not suitable for Carbon, it's better for PHP 7 codebases with scalar typehints. Take this example:

```
/** @var bool */
private $foo;

/**
 * @param bool $foo
 */
public function setFoo($foo)
{
    $this->foo = (bool) $fool
}
```

PHPStan will complain that you're casting `bool` to `bool`. But you can't remove the casting because something else can be passed there and to keep the integrity of the object, you need to be sure that only `bool` is assigned to `$this->foo`.

But in this case:

```
public function setFoo(bool $foo)
{
    $this->foo = (bool) $fool
}
```

The `(bool)` can be safely removed.